tests/test_connection.py

```diff
@@ -420,3 +424,138 @@ def test_no_mfa_passcode(mock_connect, runner):
+            "SNOWFLAKE_DATABASE": "test_database",
+            "SNOWFLAKE_WAREHOUSE": "large",
+            "SNOWFLAKE_ROLE": "role",
+            "SNOWFLAKE_SCHEMA": "my_schema",
+            "SNOWFLAKE_PASSWORD": "dummy",
+        },
+    ],
+)
+@mock.patch("snowflake.connector.connect")
+def test_connection_details_are_resolved_using_also_generic_variables(
+    mock_connect, env, test_snowcli_config, runner
+):
+    with mock.patch.dict(os.environ, env, clear=True):
+
+        result = runner.invoke(["sql", "-q", "select 1", "-c", "empty"])
+
+        assert result.exit_code == 0, result.output
+        args, kwargs = mock_connect.call_args
+        assert kwargs == {
+            "account": "some_account",
+            "application": "SNOWCLI.SQL",
+            "database": "test_database",
+            "warehouse": "large",
+            "schema": "my_schema",
+            "role": "role",
+            "password": "dummy",
+        }
+
+
+@pytest.mark.parametrize(
+    "env",
+    [
+        {
+            "SNOWFLAKE_CONNECTIONS_EMPTY_ACCOUNT": "some_account",
+            "SNOWFLAKE_CONNECTIONS_EMPTY_DATABASE": "test_database",
+            "SNOWFLAKE_CONNECTIONS_EMPTY_WAREHOUSE": "large",
+            "SNOWFLAKE_CONNECTIONS_EMPTY_ROLE": "role",
+            "SNOWFLAKE_CONNECTIONS_EMPTY_SCHEMA": "my_schema",
+        },
+        {
+            "SNOWFLAKE_ACCOUNT": "some_account",
+            "SNOWFLAKE_DATABASE": "test_database",
+            "SNOWFLAKE_WAREHOUSE": "large",
+            "SNOWFLAKE_ROLE": "role",
+            "SNOWFLAKE_SCHEMA": "my_schema",
+        },
+    ],
+)
+@mock.patch("snowflake.connector.connect")
+def test_flags_take_precedence_before_environment_variables(
+    mock_connect, env, test_snowcli_config, runner
+):
+    with mock.patch.dict(os.environ, env, clear=True):
+
+        result = runner.invoke(
+            [
+                "sql",
+                "-q",
+                "select 1",
+                "-c",
+                "empty",
+                "--account",
+                "account_from_flag",
+                "--database",
+                "database_from_flag",
+                "--schema",
+                "schema_from_flag",
+                "--password",
+                "password_from_flag",
+                "--role",
+                "role_from_flag",
+            ]
+        )
+
+        assert result.exit_code == 0, result.output
+        args, kwargs = mock_connect.call_args
+        assert kwargs == {
+            "account": "account_from_flag",
+            "application": "SNOWCLI.SQL",
+            "database": "database_from_flag",
+            "warehouse": "large",
+            "schema": "schema_from_flag",
+            "password": "password_from_flag",
+            "role": "role_from_flag",
+        }
+
+
+@mock.patch.dict(
+    os.environ,
+    {
+        "SNOWFLAKE_CONNECTIONS_EMPTY_ACCOUNT": "account_from_flag",
+        "SNOWFLAKE_ACCOUNT": "account_from_flag",
```

Comment:
can we have different values for every source:
- `account_from_flag`
- `account_from_global_env`
- `account_from_connection_env`
- `account_from_connection_config`

now when you have the same value in three sources testing it kind of doesn't make sense

this comment is not only about this variable and this test case but all variables in all your test cases